### ISSUE TITLE
Centered popup doesn't fit entire window

### ISSUE THREAD

**Issue by igordmn** [JetBrains/compose-jb#1485]
Compose 1.0.0-rc5

```
import androidx.compose.foundation.background
import androidx.compose.foundation.layout.Box
import androidx.compose.foundation.layout.fillMaxSize
import androidx.compose.foundation.layout.wrapContentSize
import androidx.compose.material.Card
import androidx.compose.material.Text
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.window.Popup
import androidx.compose.ui.window.Window
import androidx.compose.ui.window.application

fun main() = application {
    Window(onCloseRequest = ::exitApplication) {
        Box(Modifier.fillMaxSize().background(Color.Cyan)) {
            Popup(alignment = Alignment.Center) {
                Card(Modifier.fillMaxSize(1f)) {
                    Box(contentAlignment = Alignment.Center) {
                        Text("Testing!", Modifier.wrapContentSize())
                    }
                }
            }
        }
    }
}
```
Actual:
![image](https://user-images.githubusercontent.com/5963351/143775316-7618ebb9-4232-4a96-b65b-5b04a3b323bc.png)

Expected:
![image](https://user-images.githubusercontent.com/5963351/143775328-581d320a-73a6-48d2-b679-3f074b5edc5a.png)

It is probably because we changed constraints [here](https://github.com/JetBrains/androidx/pull/79).

We can fix the issue in that PR another way - instead of `max(height - position.y, position.y)` we can send `height` as before, but when the popup doesn't fit into the window, we can shift it.

**Comment by igordmn** [comment]
Should be fixed in 1.0.0-rc6

**Comment by okushnikov** [comment]
Please check the <a href="https://youtrack.jetbrains.com/issue/CMP-5145">following ticket</a> on YouTrack for follow-ups to this issue. GitHub issues will be closed in the coming weeks.
